modules/gateways/Invoice/lib/Invoice.class.php

```diff
@@ -287,7 +287,7 @@ public function send_invoice($download = false)
         $invoice_name .= esc_html__(' for ', 'event_espresso') . $template_args['name'];
         $invoice_name = str_replace(' ', '_', $invoice_name);
         // Create the PDF
-        if ($request->requestParamIsSet('html')) {
+        if ($request->requestParamIsSet('html') || ! is_readable(EE_THIRD_PARTY . 'dompdf/src/Autoloader.php')) {
```

Comment:
i get what you are doing, but I don't think this is the best way to handle this because it's essentially changing the intended result. If the `html` request param is set, then the user is not expecting a PDF document (not saying that is happening here), and conversely, the user is not requesting or expecting an HTML rendering if they have requested a PDF. So, although this change _does_ protect against the `Dompdf` library being missing, it will produce unexpected and unwanted results IMHO. 

To fix this, the `is_readable()` check needs to be moved into the the `else` block of code below this, and should likely generate an error if DOMPDF is not readable, then execute the existing code (which is not great and could be improved to be more defensive as it currently breaks the golden rule of programming (“If it can be null, it will be null”).

On a side note, the `else` block below is not necessary if we return from the previous condition after echoing; ie: 

```php
if ($request->requestParamIsSet('html')) {
    echo wp_kses($content, AllowedTags::getWithFormTags());
    return;
}
// no else block necessary here
```